File: 2026-08/day0054/f3_Ajax_POST_request.py
from flask import Flask, request, jsonify

app = Flask(__name__)

tasks = {
    "3": {
        "title": "Flask",
        "done": False
    }
}


@app.route("/tasks/<task_id>/title", methods=["POST"])
def changeTitle(task_id):
    title = request.form.get("title")

    if task_id in tasks:
        tasks[task_id]["title"] = title

        task = tasks[task_id]

        return jsonify(task)

File: 2026-08/day0054/f3_Ajax_POST_request.js
const taskId = "3";
const newTitle = "Flask 복습 완료";

$.ajax({
    url: `/tasks/${taskId}/title`,
    type: "POST",
    data: {
        title: newTitle
    },
    success: function(response) {
        console.log(response.title)
    }
})